Add label identifier to `card_m`

agustibr opened the pull request:
#### :tophat: What? Why?

`card_m` lack's the label which indentifies which type of card useful in some contexts like search.

This PR doesn't include this option in any view.

#### :pushpin: Related Issues
- Related to #3109
- Related to #3042

#### :clipboard: Subtasks
- [ ] Add `CHANGELOG` entry
- [x] Add documentation regarding the feature 

### :camera: Screenshots (optional)

![screenshot from 2018-05-21 16-40-04](https://user-images.githubusercontent.com/210216/40315614-25b070c4-5d1c-11e8-83af-3b9cf67ed315.png)

![screenshot from 2018-05-21 17-29-31](https://user-images.githubusercontent.com/210216/40315762-93dc9d3e-5d1c-11e8-97c9-9c3dae222e43.png)

mrcasals commented:
@agustibr why do you need to add all this CSS? The design app uses the CSS defined at `decidim-core`, and it has examples of cards with identifiers, so you shouldn't need to add that CSS 😕 

https://decidim-design.herokuapp.com/public/library/pattern-library (under "Tarjetas tamaño M" -> "Card con identificador")

/cc @Crashillo

agustibr commented:
@mrcasals Yes, I've looked at the _pattern library_ 

I've modified it a little bit because if the background color of the card is different from the wrapper background, you could see a rectangle below the label. So I added the `text-shadow` property to add a border to the text for the top part that comes out of the card and the `::before` to cover the border that passes behind the text, I attach screenshots:

+ Before PR: 
  ![screenshot from 2018-05-22 09-37-47](https://user-images.githubusercontent.com/210216/40348427-feeab7aa-5da3-11e8-8235-5864ddaa097d.png)
+ After:
  ![screenshot from 2018-05-22 09-38-10](https://user-images.githubusercontent.com/210216/40348458-0dee99ba-5da4-11e8-8604-6c26c710f54b.png)

https://github.com/decidim/decidim/blob/81fd4b7e62f12a249136b72bb855cedba5336ade/decidim-core/app/assets/stylesheets/decidim/modules/_cards.scss#L735

https://github.com/decidim/decidim/blob/81fd4b7e62f12a249136b72bb855cedba5336ade/decidim-core/app/assets/stylesheets/decidim/modules/_cards.scss#L744

mrcasals commented:
Oh, I see! Then I'll leave the review for this to @Crashillo 😄

Crashillo commented on decidim-core/app/assets/stylesheets/decidim/modules/_cards.scss:
```diff
@@ -730,9 +731,39 @@ $datetime-bg: $primary;
+    top: $global-margin * .5;
+  }
+
+  .success &,
```
This snippet compiles to:
```css
.success .card__label, .muted .card__label, .warning .card__label, .alert .card__label {
  margin-top: -0.95rem;
}
.success .card__label::before, .muted .card__label::before, .warning .card__label::before, .alert .card__label::before {
  height: 8px;
  top: 0.45rem;
}
```
That means it loses the card scope. It would so odd, but possible, that any success/alert/whatever status applies these styles to a `.card__label` class.
So, to solve this out, prepend the `.card` class. In that way:
```css
.card.success &,
.card.muted &,
.card.warning &,
.card.alert & {
   // properties
}
```
Now we're guaranteeing that the card status, and only that status, is which applies those styles.

agustibr replied:
Yes, good point, I'll add the `.card` scope

Crashillo commented:
Very good point, @agustibr. 👍 I'm not even realized about.

Btw, the current category has some border that the new design hasn't. Are you on it?
![imagen](https://user-images.githubusercontent.com/817526/40351167-0bb212e6-5dac-11e8-9023-64c00c5e605a.png)

mrcasals commented:
@Crashillo I'd leave the category design discussion for another PR, since it's something that affects other views (eg. the proposal page)
